src/Dropdown/MenuRoot.tsx

```diff
@@ -0,0 +1,437 @@
+  placement?: TypeAttributes.Placement8;
+
+  /** Whether or not component is disabled */
+  disabled?: boolean;
+
+  /** The style of the menu */
+  menuStyle?: React.CSSProperties;
+
+  /** A css class to apply to the Toggle DOM node */
+  toggleClassName?: string;
+
+  /** The value of the current option */
+  eventKey?: T;
+
+  /** You can use a custom element type for this toggle component */
+  toggleAs?: React.ElementType;
+
+  /** No caret variation */
+  noCaret?: boolean;
+
+  /**
+   * Open the menu and control it
+   * @deprecated
+   */
+  open?: boolean;
+
+  /** Whether Dropdown menu shows header  */
+  showHeader?: boolean;
+
+  /** Custom title */
+  renderTitle?: (children?: React.ReactNode) => React.ReactNode;
+
+  /** The callback function that the menu closes */
+  onClose?: () => void;
+
+  /** Menu Pop-up callback function */
+  onOpen?: () => void;
+
+  /** Callback function for menu state switching */
+  onToggle?: (open?: boolean) => void;
+
+  /** Selected callback function */
+  onSelect?: (eventKey: T, event: React.MouseEvent<HTMLElement>) => void;
+}
+
+export interface DropdownComponent extends RsRefForwardingComponent<'div', MenuRootProps> {
+  Item: typeof DropdownItem;
+  Menu: typeof DropdownMenu;
+}
+
+const defaultProps: Partial<MenuRootProps> = {
+  as: 'div',
+  classPrefix: 'dropdown',
+  placement: 'bottomStart',
+  trigger: 'click',
+  tabIndex: 0
+};
+
+const MenuRoot: DropdownComponent = (React.forwardRef((props: MenuRootProps, ref) => {
+  const {
+    as: Component,
+    title,
+    children,
+    className,
+    menuStyle,
+    disabled,
+    renderTitle,
+    classPrefix,
+    placement,
+    activeKey,
+    toggleClassName,
+    trigger,
+    icon,
+    eventKey,
+    toggleAs,
+    noCaret,
+    style,
+    showHeader,
+    onClick,
+    onMouseEnter,
+    onMouseLeave,
+    onContextMenu,
+    onSelect,
+    onOpen,
+    onClose,
+    onToggle,
+    ...rest
+  } = props;
+
+  const { onOpenChange, sidenav, expanded } = useContext<SidenavContextType>(SidenavContext) || {};
+  const overlayTarget = useRef<HTMLUListElement>();
+  const triggerTarget = useRef<HTMLButtonElement>();
+  const menuControl = useMenuControl(overlayTarget);
+
+  const open = menuControl.open;
+  const { merge, withClassPrefix, prefix } = useClassNames(classPrefix);
+  const collapsible = sidenav && expanded;
+
+  const buttonId = useUniqueId(prefix`button-`);
+  const menuId = useUniqueId(prefix`menu-`);
+
+  const handleToggle = useCallback(
+    (isOpen?: boolean) => {
+      const nextOpen = typeof isOpen === 'undefined' ? !open : isOpen;
+      const fn = nextOpen ? onOpen : onClose;
+
+      fn?.();
+      onToggle?.(nextOpen);
+      if (nextOpen) {
+        menuControl.openMenu();
+      } else {
+        menuControl.closeMenu();
+      }
+    },
+    [
+      onClose,
+      onOpen,
+      onToggle,
+      open,
+      menuControl.focusItemAt,
+      menuControl.openMenu,
+      menuControl.closeMenu
```

Comment:
React Hook useCallback has a missing dependency: 'menuControl'. Either include it or remove the dependency array